File: 2022/03/2.py
data = open("i.txt").read().split("\n")

n = 0
for j in range(0, len(data), 3):
    i = set(data[j]).intersection(data[j+1])
    i2 = list(i.intersection(data[j+2]))[0]

    if ord(i2)< 97:
        n+= ord(i2)-65+27
    else:
        n+= ord(i2)-96

print(n)

File: 2022/03/js/2.js
const fs = require("fs");
let data = fs.readFileSync("i.txt").toString().split(/\r?\n/);
console.log(data.reduce((a1, d, i) => {
    if (i % 3 !== 0) return a1;
    return a1 + d.split("").filter(c => data[i + 1].indexOf(c) !== -1).reduce((a2, c) => {
        if (a2 !== 0 || data[i + 2].indexOf(c) === -1) return a2;
        if (c.charCodeAt(0) < 97) return c.charCodeAt(0) - 65 + 27;
        return c.charCodeAt(0) - 96;
    }, 0)
}, 0));